option java_package = "com.jbull.hermes";
//option optimize_for = LITE_RUNTIME;
message SetupMessage {
  required int32 msgNum = 1;
  optional int32 version = 2 [default = 1];
  optional string applicationName = 3 [default = "HermesIO"];
}

message SmsMessage {
  required int32 msgNum = 1;
  required int64 timeStamp = 2;
  required Contact sender = 3;
  required string content = 4;
  repeated Contact recipents = 5;
  optional bool notify = 6 [default = true];
}

message Contact {
  required int32 msgNum = 1;
  required string phoneNumber = 2;
  optional string name = 3;
  optional bytes image = 4;
}

message BatchContacts {
  required int32 msgNum = 1;
  repeated Contact contacts = 2;
}

message Mode {
  required int32 msgNum = 1;
  required int64 lastUpdate = 2; //time since last update. server time
  required int64 currentTimestamp = 3;
  required bool serverSend = 4;
}

message SyncContacts {
  required int32 msgNum = 1;
}

message Header {
  required fixed32 msgNum = 1;
  required fixed32 length = 2;
  enum Type {
    SETUPMESSAGE = 0;
    SMSMESSAGE = 1;
    CONTACT = 2;
    MODE = 3;
    SYNCCONTACTS = 4;
    BATCHCONTACTS = 5;
  }
  required Type type = 3;
}

message Ack {
  required int32 msgNum = 1;
}
